dipy/denoise/localpca.py

```diff
@@ -35,7 +36,13 @@ def _pca_classifier(L, nvoxels):
 
     Notes
     -----
-    This is based on the algorithm described in [1]_.
+    This is based on the algorithm described in [1]_. Note that the
+    condition expressed in the while-loop is expressed in terms of the
+    variance of equation (12), not equation (11) as in [1]_. Also,
+    this code implements ascending eigenvalues, unlike [1]_.
+    Also unlike [1]_, the data matrix X is assumed to be centered,
+    therefore nvoxels has been replaced with nvoxels - 1 to account
+    for the lost degree of freedom.
```

Comment:
Although I had to remind myself of this, I think it is fairly common knowledge that subtracting the mean from the data matrix will remove a degree of freedom, such that the number of non-zero eigenvalues that can be found is reduced by one.  As a quick example: https://stats.stackexchange.com/questions/123318/why-are-there-only-n-1-principal-components-for-n-data-if-the-number-of-dime I think the alternative is to not centre the data matrix, which appears to be the case in the original publications, but probably it's better to centre the data in general. So either we centre the data and make an attempt to correct for the lost degree of freedom, which differs from the original publication, or we centre the data and make no attempt to correct, which also differs from the original publication but seems inconsistent to me.

Reply:
Yes I understand that. But it is not clear for me why this loss of degree of freedom manifests on parameter Gamma = N/M in the MP distribution (line 66).

Reply:
PS: I am referring to the parameter gamma in veraart's MP-PCA manuscript

Reply:
Reviewing the original MP paper by Marcenko and Pastur, MP distribution is defined for M, N -> inf so that gamma = N/M, so the discussion on this may be irrelevant. But, anyway, if you are not sure about it, I will recommend to leave it without removing anything to keep it consistent to the original article.

Reply:
I think you are right @RafaelNH I'll return the value to the original (the test to first discard zero eigenvalues, which uses N - 1, should remain though).